EpikCord/user.py

```diff
@@ -1,39 +1,27 @@
-from typing import Optional
-
 from discord_typings import UserData
 
 from .flags import UserFlags
-from .utils import Locale, PremiumType
+from .utils import Locale, PremiumType, instance_or_none, int_or_none
 
 
 class User:
     def __init__(self, data: UserData):
-        self.id: int = int(data["id"])
-        self.username: str = data["username"]
-        self.discriminator: str = data["discriminator"]
-        self.avatar: Optional[str] = data.get("avatar")
-        self.bot: Optional[bool] = data.get("bot")
-        self.system: Optional[bool] = data.get("system")
-        self.mfa_enabled: Optional[bool] = data.get("mfa_enabled")
-        self.banner: Optional[str] = data.get("banner")
-        self.accent_color: Optional[int] = (
-            int(data["accent_color"])
-            if "accent_color" in data and data["accent_color"]
-            else None
-        )
-        self.locale: Optional[Locale] = (
-            Locale(data["locale"]) if "locale" in data else None
-        )
-        self.verified: Optional[bool] = data.get("verified")
-        self.email: Optional[str] = data.get("email")
-        self.flags: Optional[UserFlags] = (
-            UserFlags(data["flags"]) if "flags" in data else None
-        )
-        self.premium_type: Optional[PremiumType] = (
-            PremiumType(data["premium_type"])
-            if "premium_type" in data
-            else None
+        self.id = int(data["id"])
```

Comment:
Are the typehints not required?

Reply:
Typehints are not required because they are inferred from the typed dict